src/librustc_resolve/diagnostics.rs

```diff
@@ -272,6 +272,81 @@ See the 'Use Declarations' section of the reference for more information
 on this topic:
 
 http://doc.rust-lang.org/reference.html#use-declarations
+"##,
+
+E0407: r##"
+Trait impls can only declare methods that are members of the trait in question.
```

Comment:
nit: declare isn't quite right here. the trait itself "declares" the methods, and the impl implements them.